Assets/Scripts/OnHitWithEnemy.cs

```diff
@@ -4,16 +4,21 @@
 
 public class OnHitWithEnemy : MonoBehaviour {
 
-  public string with = "enemy";
-  public int damageAmount = 2;
-  public StaminaBar staminaBar;
+    public string with = "enemy";
+    public int damageAmount = 2;
+    public StaminaBar staminaBar;
+
+    public int DamageScore;
+
+
 	void OnTriggerEnter2D(Collider2D theObject)
     {
         //if an enemy hits the attack area it is destroyed
 		if (theObject.CompareTag(with))
         {
-          theObject.gameObject.GetComponent<Health> ().DealDamage (damageAmount);
-          staminaBar.attack();
+            theObject.gameObject.GetComponent<Health> ().DealDamage (damageAmount);
+            staminaBar.attack();
+            GameObject.Find("ScoreKeeper").SendMessage("Add", DamageScore);
```

Comment:
to deal damage, health component is **required**. But to deal damage, score component is **optional**